File: src/test/java/com/benjaminsproule/swagger/gradleplugin/test/java/jaxrs/ExtendedTestResourceWithoutClassAnnotation.java
package com.benjaminsproule.swagger.gradleplugin.test.java.jaxrs;

import io.swagger.annotations.ApiOperation;

import javax.ws.rs.GET;
import javax.ws.rs.Path;

public class ExtendedTestResourceWithoutClassAnnotation extends TestResourceWithoutClassAnnotation {

    @ApiOperation("An overriden operation description")
    @Path("/root/withoutannotation/overriden")
    @GET
    @Override
    public String overriden() {
        return "";
    }

    @Path("/root/withoutannotation/overridenWithoutDescription")
    @GET
    public String overridenWithoutDescription() {
        return "";
    }
}

File: scala-tests/src/main/scala/com/benjaminsproule/swagger/gradleplugin/test/scala/jaxrs/ExtendedTestResourceWithoutClassAnnotation.scala
package com.benjaminsproule.swagger.gradleplugin.test.scala.jaxrs

import io.swagger.annotations.ApiOperation
import javax.ws.rs.{GET, Path}

class ExtendedTestResourceWithoutClassAnnotation extends TestResourceWithoutClassAnnotation {

    @ApiOperation("An overriden operation description")
    @Path("/root/withoutannotation/overriden")
    @GET
    @Override
    override def overriden(): String = {
        ""
    }

    @Path("/root/withoutannotation/overridenWithoutDescription")
    @GET
    override def overridenWithoutDescription(): String = {
        ""
    }
}
